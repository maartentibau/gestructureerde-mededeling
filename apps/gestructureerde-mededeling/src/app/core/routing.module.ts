import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { CreateComponent } from '../components/create/create.component';
import { GenerateComponent } from '../components/generate/generate.component';
import { ValidateComponent } from '../components/validate/validate.component';

const routes: Routes = [
  { path: 'create', component: CreateComponent },
  { path: 'generate', component: GenerateComponent },
  { path: 'validate', component: ValidateComponent },
  { path: '', redirectTo: '/generate', pathMatch: 'full' },
  { path: '**', component: GenerateComponent },
];

@NgModule({
  imports: [RouterModule.forRoot(routes, { relativeLinkResolution: 'legacy' })],
  exports: [RouterModule],
})
export class RoutingModule {}
