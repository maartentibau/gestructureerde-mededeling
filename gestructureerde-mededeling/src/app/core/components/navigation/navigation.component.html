<mat-toolbar>
  <mat-toolbar-row>
    <h1>
      <a mat-button routerLink="/">{{ appTitle$ | async }}</a>
    </h1>
    <span class="flex-spacer"></span>
    <nav>
      <ng-container *ngIf="isSmallScreen$ | async; then smallNavigation; else largeNavigation"></ng-container>
    </nav>
  </mat-toolbar-row>
</mat-toolbar>

<ng-template #smallNavigation>
  <a
    *ngFor="let item of navigation$ | async; index as i"
    [attr.data-navigation-id]="i"
    mat-icon-button
    routerLinkActive="mat-mdc-raised-button"
    [routerLink]="item.path"
    ><fa-icon [icon]="item.icon"></fa-icon
  ></a>
</ng-template>

<ng-template #largeNavigation>
  <a
    *ngFor="let item of navigation$ | async; index as i"
    [attr.data-navigation-id]="i"
    mat-button
    routerLinkActive="mat-mdc-raised-button"
    [routerLink]="item.path"
    ><fa-icon [icon]="item.icon"></fa-icon>{{ item.label }}</a
  >
</ng-template>
