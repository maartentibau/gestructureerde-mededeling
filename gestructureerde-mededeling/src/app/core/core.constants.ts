export const DEFAULT_TITLE = `Gestructureerde Mededeling - OGM`;
