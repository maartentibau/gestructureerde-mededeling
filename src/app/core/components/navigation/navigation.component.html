<mat-toolbar>
  <mat-toolbar-row>
    <h1>
      <a mat-button routerLink="/" title="Gestructureerde Mededeling" aria-label="Ga terug naar de homepage">{{
        appTitle$ | async
      }}</a>
    </h1>
    <span class="flex-spacer"></span>
    <nav>
      @if (isSmallScreen$ | async) {
        @for (item of navigation$ | async; track item; let i = $index) {
          <a
            [attr.data-navigation-id]="i"
            mat-icon-button
            routerLinkActive="mat-mdc-raised-button"
            [routerLink]="item.path"
            [title]="item.title"
            ><fa-icon [icon]="item.icon"></fa-icon
          ></a>
        }
      } @else {
        @for (item of navigation$ | async; track item; let i = $index) {
          <a [attr.data-navigation-id]="i" mat-button routerLinkActive="mat-mdc-raised-button" [routerLink]="item.path"
            ><fa-icon [icon]="item.icon"></fa-icon>{{ item.label }}</a
          >
        }
      }
    </nav>
  </mat-toolbar-row>
</mat-toolbar>
