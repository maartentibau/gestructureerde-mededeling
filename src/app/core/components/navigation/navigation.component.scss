mat-toolbar {
  background-color: #0c652b;
}

h1 {
  color: #fff;
}

nav {
  color: #fff;

  a {
    text-transform: uppercase;
    margin: 0 0.5em;

    &.mdc-button > ::ng-deep span.mdc-button__label {
      display: flex;
      justify-content: center;
      align-items: center;

      & > fa-icon {
        margin-right: 8px;
      }
    }

    &:hover:not(.mat-mdc-raised-button):not(.active) {
      background-color: #033313;
    }

    &.mat-mdc-raised-button {
      color: #216a38;

      &:hover {
        background-color: white;
      }
    }

    fa-icon {
      font-size: 1rem;
    }
  }
}
