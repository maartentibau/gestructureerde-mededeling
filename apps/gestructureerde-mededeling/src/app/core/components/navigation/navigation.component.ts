import { ChangeDetectionStrategy, Component } from '@angular/core';
import { NavigationEntity, NavigationLabel } from './navigation.model';
import { Observable } from 'rxjs';
import { map, tap } from 'rxjs/operators';
import { Breakpoints } from '@angular/cdk/layout';
import { ScreenService } from '../../services/screen.service';
import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
import { faCog, faShieldCheck, faStar } from '@fortawesome/pro-solid-svg-icons';

export const APP_NAVIGATION: NavigationEntity[] = [
  { path: ['/create'], label: { long: 'Zelf mededeling maken', short: 'Zelf maken' }, icon: ['fas', 'star'] },
  { path: ['/generate'], label: { long: 'Genereer mededeling', short: 'Genereren' }, icon: ['fas', 'cog'] },
  {
    path: ['/validate'],
    label: { long: 'Controleer mededeling', short: 'Controleren' },
    icon: ['fas', 'shield-check'],
  },
];

@Component({
  selector: 'ogm-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NavigationComponent {
  readonly navigation$: Observable<Omit<NavigationEntity, 'label'>[]>;
  readonly isSmallScreen$: Observable<boolean>;
  readonly appTitle$: Observable<string>;

  constructor(private screenService: ScreenService, private faIconLibrary: FaIconLibrary) {
    this.isSmallScreen$ = this.screenService
      .observerBreakpoints()
      .pipe(
        map((breakpoints: { [key: string]: boolean }) =>
          Boolean(breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small])
        )
      );

    this.navigation$ = this.screenService.observerBreakpoints().pipe(
      map((breakpoints: { [key: string]: boolean }) => {
        const labelSize: NavigationLabel = Boolean(
          breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small] || breakpoints[Breakpoints.Medium]
        )
          ? NavigationLabel.Short
          : NavigationLabel.Long;

        return APP_NAVIGATION.map((navigation: NavigationEntity) => ({
          ...navigation,
          label: navigation.label[labelSize],
        }));
      })
    );

    this.appTitle$ = this.screenService.observerBreakpoints().pipe(
      tap(console.log),
      map((breakpoints: { [key: string]: boolean }) =>
        breakpoints[Breakpoints.XSmall] ? 'OGM' : 'Gestructureerde Mededeling'
      )
    );

    this.faIconLibrary.addIcons(faShieldCheck, faCog, faStar);
  }
}
