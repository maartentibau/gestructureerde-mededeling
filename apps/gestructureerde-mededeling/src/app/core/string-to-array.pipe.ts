import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
  name: 'stringToArray',
})
export class StringToArrayPipe implements PipeTransform {
  transform(value: string | null | undefined): string[] {
    if ((value && typeof value === 'string') || typeof value === 'number') {
      return Array.from(String(value));
    }

    return [];
  }
}
