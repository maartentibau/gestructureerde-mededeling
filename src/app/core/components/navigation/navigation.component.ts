import { Breakpoints } from '@angular/cdk/layout';
import { AsyncPipe } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatToolbarModule } from '@angular/material/toolbar';
import { RouterLink, RouterLinkActive } from '@angular/router';
import { FaIconLibrary, FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { faCheckCircle, faCog, faStar } from '@fortawesome/free-solid-svg-icons';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { ScreenService } from '../../services/screen.service';
import { MainNavigation, NavigationEntity, NavigationLabel } from './navigation.model';

export const APP_NAVIGATION: NavigationEntity[] = [
  { path: ['/create'], label: { long: 'Zelf mededeling maken', short: 'Zelf maken' }, icon: ['fas', 'star'] },
  { path: ['/generate'], label: { long: 'Genereer mededeling', short: 'Genereren' }, icon: ['fas', 'cog'] },
  {
    path: ['/validate'],
    label: { long: 'Controleer mededeling', short: 'Controleren' },
    icon: ['fas', 'check-circle'],
  },
];

@Component({
  imports: [
    MatToolbarModule,
    MatButtonModule,
    AsyncPipe,
    FontAwesomeModule,
    MatButtonModule,
    RouterLink,
    RouterLinkActive,
  ],
  selector: 'ogm-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NavigationComponent {
  readonly navigation$: Observable<MainNavigation>;
  readonly isSmallScreen$: Observable<boolean>;
  readonly appTitle$: Observable<string>;

  constructor(
    private screenService: ScreenService,
    private faIconLibrary: FaIconLibrary,
  ) {
    this.isSmallScreen$ = this.screenService
      .observerBreakpoints()
      .pipe(
        map((breakpoints: { [key: string]: boolean }) =>
          Boolean(breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small]),
        ),
      );

    this.navigation$ = this.screenService.observerBreakpoints().pipe(
      map((breakpoints: { [key: string]: boolean }) => {
        const labelSize: NavigationLabel =
          breakpoints[Breakpoints.XSmall] || breakpoints[Breakpoints.Small] || breakpoints[Breakpoints.Medium]
            ? NavigationLabel.Short
            : NavigationLabel.Long;

        return APP_NAVIGATION.map((navigation: NavigationEntity) => ({
          ...navigation,
          label: navigation.label[labelSize],
        }));
      }),
    );

    this.appTitle$ = this.screenService
      .observerBreakpoints()
      .pipe(
        map((breakpoints: { [key: string]: boolean }) =>
          breakpoints[Breakpoints.XSmall] ? 'OGM' : 'Gestructureerde Mededeling - OGM',
        ),
      );

    this.faIconLibrary.addIcons(faCheckCircle, faCog, faStar);
  }
}
