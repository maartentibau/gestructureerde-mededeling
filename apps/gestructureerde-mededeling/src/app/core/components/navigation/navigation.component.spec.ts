import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { APP_NAVIGATION, NavigationComponent } from './navigation.component';
import { RouterTestingModule } from '@angular/router/testing';
import { Breakpoints } from '@angular/cdk/layout';
import { MaterialModule } from '../../material.module';
import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
import { ScreenService } from '../../services/screen.service';
import { TestScheduler } from 'rxjs/testing';
import { NavigationEntity, NavigationLabel } from './navigation.model';

describe('NavigationComponent', () => {
  let component: NavigationComponent;
  let fixture: ComponentFixture<NavigationComponent>;

  let screenService: ScreenService;
  let testScheduler: TestScheduler;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule, MaterialModule, FontAwesomeModule],
      declarations: [NavigationComponent],
      providers: [ScreenService],
    }).compileComponents();
  }));

  beforeEach(() => {
    screenService = TestBed.inject(ScreenService);

    fixture = TestBed.createComponent(NavigationComponent);
    component = fixture.componentInstance;

    testScheduler = new TestScheduler((actual, expected) => {
      expect(actual).toEqual(expected);
    });
  });

  it('should create', () => {
    // act
    fixture.detectChanges();

    // assert
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  describe('isSmallScreen$', () => {
    [
      { breakpoint: Breakpoints.XSmall, smallScreen: true },
      { breakpoint: Breakpoints.Small, smallScreen: true },
      { breakpoint: Breakpoints.Medium, smallScreen: false },
      { breakpoint: Breakpoints.Large, smallScreen: false },
      { breakpoint: Breakpoints.XLarge, smallScreen: false },
    ].forEach(({ breakpoint, smallScreen }) => {
      it(`should emit ${smallScreen} when ${breakpoint}`, () => {
        testScheduler.run(({ cold, expectObservable }) => {
          // prepare
          const breakPoints$ = cold('a', { a: { [breakpoint]: true } });
          jest.spyOn(screenService, 'observerBreakpoints').mockReturnValue(breakPoints$);

          // Because the component uses the screenService.observerBreakpoints stream in it's constructor
          // we're ensuring the mocks are set up before creating the component.
          // act
          fixture = TestBed.createComponent(NavigationComponent);
          component = fixture.componentInstance;

          // check
          expectObservable(component.isSmallScreen$).toBe('a', { a: smallScreen });
        });
      });
    });
  });

  describe('navigation$', () => {
    [
      { breakpoint: Breakpoints.XSmall, labelSize: NavigationLabel.Short },
      { breakpoint: Breakpoints.Small, labelSize: NavigationLabel.Short },
      { breakpoint: Breakpoints.Medium, labelSize: NavigationLabel.Short },
      { breakpoint: Breakpoints.Large, labelSize: NavigationLabel.Long },
      { breakpoint: Breakpoints.XLarge, labelSize: NavigationLabel.Long },
    ].forEach(({ breakpoint, labelSize }) => {
      it(`should use label ${labelSize} when ${breakpoint}`, () => {
        testScheduler.run(({ cold, expectObservable }) => {
          // prepare
          const expectedNavigation = APP_NAVIGATION.map((navigation: NavigationEntity) => ({
            ...navigation,
            label: navigation.label[labelSize],
          }));

          const breakPoints$ = cold('a', { a: { [breakpoint]: true } });
          jest.spyOn(screenService, 'observerBreakpoints').mockReturnValue(breakPoints$);

          // Because the component uses the screenService.observerBreakpoints stream in it's constructor
          // we're ensuring the mocks are set up before creating the component.
          // act
          fixture = TestBed.createComponent(NavigationComponent);
          component = fixture.componentInstance;

          // check
          expectObservable(component.navigation$).toBe('a', { a: expectedNavigation });
        });
      });
    });
  });

  describe('appTitle$', () => {
    [
      { breakpoint: Breakpoints.XSmall, title: 'OGM' },
      { breakpoint: Breakpoints.Small, title: 'Gestructureerde Mededeling' },
      { breakpoint: Breakpoints.Medium, title: 'Gestructureerde Mededeling' },
      { breakpoint: Breakpoints.Large, title: 'Gestructureerde Mededeling' },
      { breakpoint: Breakpoints.XLarge, title: 'Gestructureerde Mededeling' },
    ].forEach(({ breakpoint, title }) => {
      it(`should be ${title} when ${breakpoint}`, () => {
        testScheduler.run(({ cold, expectObservable }) => {
          // prepare
          const breakPoints$ = cold('a', { a: { [breakpoint]: true } });
          jest.spyOn(screenService, 'observerBreakpoints').mockReturnValue(breakPoints$);

          // Because the component uses the screenService.observerBreakpoints stream in it's constructor
          // we're ensuring the mocks are set up before creating the component.
          // act
          fixture = TestBed.createComponent(NavigationComponent);
          component = fixture.componentInstance;

          // check
          expectObservable(component.appTitle$).toBe('a', { a: title });
        });
      });
    });
  });
});
