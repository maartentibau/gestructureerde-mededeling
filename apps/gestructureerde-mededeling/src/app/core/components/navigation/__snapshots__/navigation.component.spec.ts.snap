// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationComponent should create 1`] = `
<ogm-navigation
  __ngContext__="0"
  appTitle$={[Function Observable]}
  faIconLibrary={[Function FaIconLibrary]}
  isSmallScreen$={[Function Observable]}
  navigation$={[Function Observable]}
  screenService={[Function ScreenService]}
>
  <mat-toolbar
    class="mat-toolbar mat-toolbar-multiple-rows"
  >
    <mat-toolbar-row
      class="mat-toolbar-row"
    >
      <h1>
        <a
          aria-disabled="false"
          class="mat-focus-indicator mat-button mat-button-base"
          href="/"
          mat-button=""
          routerlink="/"
        >
          <span
            class="mat-button-wrapper"
          >
            Gestructureerde Mededeling
          </span>
          <span
            class="mat-ripple mat-button-ripple"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </a>
      </h1>
      <span
        class="flex-spacer"
      />
      <nav>
        <a
          aria-disabled="false"
          class="mat-focus-indicator mat-button mat-button-base"
          data-navigation-id="0"
          href="/create"
          mat-button=""
          routerlinkactive="mat-raised-button"
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-star"
                data-icon="star"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
            Zelf mededeling maken
          </span>
          <span
            class="mat-ripple mat-button-ripple"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </a>
        <a
          aria-disabled="false"
          class="mat-focus-indicator mat-button mat-button-base"
          data-navigation-id="1"
          href="/generate"
          mat-button=""
          routerlinkactive="mat-raised-button"
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-gear"
                data-icon="gear"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M495.9 166.6C499.2 175.2 496.4 184.9 489.6 191.2L446.3 230.6C447.4 238.9 448 247.4 448 256C448 264.6 447.4 273.1 446.3 281.4L489.6 320.8C496.4 327.1 499.2 336.8 495.9 345.4C491.5 357.3 486.2 368.8 480.2 379.7L475.5 387.8C468.9 398.8 461.5 409.2 453.4 419.1C447.4 426.2 437.7 428.7 428.9 425.9L373.2 408.1C359.8 418.4 344.1 427 329.2 433.6L316.7 490.7C314.7 499.7 307.7 506.1 298.5 508.5C284.7 510.8 270.5 512 255.1 512C241.5 512 227.3 510.8 213.5 508.5C204.3 506.1 197.3 499.7 195.3 490.7L182.8 433.6C167 427 152.2 418.4 138.8 408.1L83.14 425.9C74.3 428.7 64.55 426.2 58.63 419.1C50.52 409.2 43.12 398.8 36.52 387.8L31.84 379.7C25.77 368.8 20.49 357.3 16.06 345.4C12.82 336.8 15.55 327.1 22.41 320.8L65.67 281.4C64.57 273.1 64 264.6 64 256C64 247.4 64.57 238.9 65.67 230.6L22.41 191.2C15.55 184.9 12.82 175.3 16.06 166.6C20.49 154.7 25.78 143.2 31.84 132.3L36.51 124.2C43.12 113.2 50.52 102.8 58.63 92.95C64.55 85.8 74.3 83.32 83.14 86.14L138.8 103.9C152.2 93.56 167 84.96 182.8 78.43L195.3 21.33C197.3 12.25 204.3 5.04 213.5 3.51C227.3 1.201 241.5 0 256 0C270.5 0 284.7 1.201 298.5 3.51C307.7 5.04 314.7 12.25 316.7 21.33L329.2 78.43C344.1 84.96 359.8 93.56 373.2 103.9L428.9 86.14C437.7 83.32 447.4 85.8 453.4 92.95C461.5 102.8 468.9 113.2 475.5 124.2L480.2 132.3C486.2 143.2 491.5 154.7 495.9 166.6V166.6zM256 336C300.2 336 336 300.2 336 255.1C336 211.8 300.2 175.1 256 175.1C211.8 175.1 176 211.8 176 255.1C176 300.2 211.8 336 256 336z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
            Genereer mededeling
          </span>
          <span
            class="mat-ripple mat-button-ripple"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </a>
        <a
          aria-disabled="false"
          class="mat-focus-indicator mat-button mat-button-base"
          data-navigation-id="2"
          href="/validate"
          mat-button=""
          routerlinkactive="mat-raised-button"
        >
          <span
            class="mat-button-wrapper"
          >
            <fa-icon
              class="ng-fa-icon"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle-check"
                data-icon="circle-check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM371.8 211.8C382.7 200.9 382.7 183.1 371.8 172.2C360.9 161.3 343.1 161.3 332.2 172.2L224 280.4L179.8 236.2C168.9 225.3 151.1 225.3 140.2 236.2C129.3 247.1 129.3 264.9 140.2 275.8L204.2 339.8C215.1 350.7 232.9 350.7 243.8 339.8L371.8 211.8z"
                  fill="currentColor"
                />
              </svg>
            </fa-icon>
            Controleer mededeling
          </span>
          <span
            class="mat-ripple mat-button-ripple"
            matripple=""
          />
          <span
            class="mat-button-focus-overlay"
          />
        </a>
      </nav>
    </mat-toolbar-row>
  </mat-toolbar>
</ogm-navigation>
`;
